hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -977,6 +988,16 @@ public String getKeyGeneratorClass() {
     return getString(KEYGENERATOR_CLASS_NAME);
   }
 
+  public HoodieRecord.HoodieRecordType getRecordType() {
+    HoodieRecordType recordType = HoodieRecord.HoodieRecordType.valueOf(getString(RECORD_TYPE));
+    String basePath = getString(BASE_PATH);
+    boolean metadataTable = HoodieTableMetadata.isMetadataTable(basePath);
+    if (metadataTable) {
+      recordType = HoodieRecordType.AVRO;
+    }
+    return recordType;
```

Comment:
looks like this `getRecordType()` is invoked everywhere throughout the write path. we should optimize the perf here somehow. for e.g., the ENUM instantiation , metadata table check. since this info is not gonna change throughout a write operation, we should consider cache it.

Reply:
Fixed. Cached the HoodieRecordType.